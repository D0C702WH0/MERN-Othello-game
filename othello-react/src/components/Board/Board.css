.board {
  flex-wrap: nowrap;
  max-width: 60rem;
  max-height: 60rem;
  margin: 0 auto;
}

.board_container {
  display: flex;
}

.player_container {
  display: flex;
  flex-direction: column;
}

.no_gutter {
  padding-left: 0;
  padding-right: 0;
  display: inline-block;
}

.pawn_player {
  margin: 0 auto;
  max-height: 50px;
  max-width: 50px;
}