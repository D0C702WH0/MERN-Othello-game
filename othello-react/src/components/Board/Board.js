import React from 'react'
import Column from '../Column/Column'
import PropTypes from 'prop-types'
import './Board.css'

export default function Board ({ board, click }) {
  return (
    <div className="board_container">
      <div className="board">
        {board._squares.map((col, i) =>
          <Column key={i} click={click} col={col} />
        )}
      </div>
    </div>
  )
}

Board.propTypes = {
  board: PropTypes.object,
  click: PropTypes.func,
  pieceType: PropTypes.string
}
