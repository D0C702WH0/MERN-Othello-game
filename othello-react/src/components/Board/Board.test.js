import React from 'react'
import TestRenderer from 'react-test-renderer'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Board from './Board'
import reversi from 'reversi'

const GameGameInstance = reversi.Game
const game = new GameGameInstance()

configure({ adapter: new Adapter() })

let renderer
const handleClick = jest.fn()
beforeEach(() => {
  renderer = mount(
    <Board board={game._board} click={handleClick}/>
  )
})

describe('Board', () => {
  it('Basic rendering', () => {
    const renderer = TestRenderer.create(
      <Board board={game._board} click={handleClick}/>
    )
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('Board has 8 Rows', () => {
    expect(renderer.find('.row')).toHaveLength(8)
  })

  it('Board has 64 Squares', () => {
    expect(renderer.find('.square')).toHaveLength(64)
  })

  it('Board has 2 black pawns', () => {
    expect(renderer.find('.black')).toHaveLength(2)
  })

  it('Board has 2 white pawns', () => {
    expect(renderer.find('.white')).toHaveLength(2)
  })
})
