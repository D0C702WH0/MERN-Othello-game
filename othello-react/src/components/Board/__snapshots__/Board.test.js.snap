// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Basic rendering 1`] = `
<div
  className="board_container"
>
  <div
    className="board"
  >
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn white"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn black"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn black"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn white"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
      <div
        className="zoom"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
      >
        <div
          className="square"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={-1}
        >
          <div
            className="pawn undefined"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
